deploy/buildtools/buildafi.py

```diff
@@ -84,51 +114,120 @@ def on_build_failure():
+                  exclude='build/checkpoints',
+                  ssh_opts="-o StrictHostKeyChecking=no",
+                  extra_opts="-l", capture=True)
+            rootLogger.debug(rsync_cap)
+            rootLogger.debug(rsync_cap.stderr)
+            # sync the generated_dir (in a clean build, it is only the replace-rtl.sh)
+            rsync_cap = rsync_project(local_dir=pjoin(generated_dir, ''),
+                          remote_dir=generated_dir,
+                          ssh_opts="-o StrictHostKeyChecking=no",
+                          extra_opts="-l", capture=True)
+            rootLogger.debug(rsync_cap)
+            rootLogger.debug(rsync_cap.stderr)
+            # sync all components of the generator classpath
+            # classpath members are directories or archives
+            for e in gen_classpath.split(":"):
+                run('mkdir -p ' + dirname(e))
+                rsync_cap = rsync_project(local_dir=e,
+                              remote_dir=e,
+                              ssh_opts="-o StrictHostKeyChecking=no",
+                              extra_opts="-l", capture=True)
+                rootLogger.debug(rsync_cap)
+                rootLogger.debug(rsync_cap.stderr)
+        elab_result = 0
+        with InfoStreamLogger('stdout'), InfoStreamLogger('stderr'), settings(warn_only=True):
+            # install java and dtc on the build host
+            sudo('yum install -y java dtc')
+            # run the replace-rtl.sh script
+            rootLogger.info("Running Verilog generation via: " + pjoin(generated_dir,'replace-rtl.sh'))
+            elab_result = run('bash -xe ' + pjoin(generated_dir,'replace-rtl.sh')).return_code
+        with StreamLogger('stdout'), StreamLogger('stderr'):
+            # rsync generated_dir back to manager
+            rsync_cap = rsync_project(local_dir=generated_dir,
+                          remote_dir=pjoin(generated_dir, ''),
+                          ssh_opts="-o StrictHostKeyChecking=no", upload=False, extra_opts="-l",
+                          capture=True)
+            rootLogger.debug(rsync_cap)
+            rootLogger.debug(rsync_cap.stderr)
+            # rsync fpgabuilddir to manager
+            rsync_cap = rsync_project(local_dir=pjoin(aws_fpga_root, fpgabuilddir),
+                                      remote_dir=pjoin(aws_fpga_root, fpgabuilddir, ''),
+                                      ssh_opts="-o StrictHostKeyChecking=no", upload=False, extra_opts="-l",
+                                      capture=True)
+            rootLogger.debug(rsync_cap)
+            rootLogger.debug(rsync_cap.stderr)
+            local("mkdir -p {}/results-build/{}/".format(ddir, results_builddir))
+            local("cp {}/design/cl_firesim_generated.sv {}/results-build/{}/cl_firesim_generated.sv".format(
+                  pjoin(aws_fpga_root, fpgabuilddir),
+                  ddir,
+                  results_builddir))
```

Comment:
Testing the latest code today, depending on how `replace-rtl.sh` fails, it is possible for the `cl_firesim_generated.sv` to not have been created.  In those failure modes, we will bomb out here and the build hosts are still left running.  I'm thinking that I will put this local behind  `file-existance or elab_result != 0` so that if we're in the process of bailing out after a failure, we don't bail out too early.